cmd/minishift/cmd/config/view.go

```diff
@@ -61,33 +66,29 @@ func init() {
 	ConfigCmd.AddCommand(configViewCmd)
 }
 
-func configView() error {
-	cfg, err := ReadConfig()
+func determineTemplate(tempFormat string) (tmpl *template.Template) {
+	tmpl, err := template.New("view").Parse(tempFormat)
 	if err != nil {
-		return err
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating view template: %s", err.Error()))
 	}
-	var buffer bytes.Buffer
+	return tmpl
+}
+
+func configView(cfg MinishiftConfig, tmpl *template.Template, writer io.Writer) error {
+	var lines []string
 	for k, v := range cfg {
 		_, excluded := excludedConfigKeys[k]
 		if excluded {
 			continue
 		}
-
-		tmpl, err := template.New("view").Parse(configViewFormat)
-		if err != nil {
-			atexit.ExitWithMessage(1, fmt.Sprintf("Error creating view template: %s", err.Error()))
-		}
 		viewTmplt := ConfigViewTemplate{k, v}
-		err = tmpl.Execute(&buffer, viewTmplt)
-		if err != nil {
-			atexit.ExitWithMessage(1, fmt.Sprintf("Error executing view template: %s", err.Error()))
+		var buffer bytes.Buffer
+		if err := tmpl.Execute(&buffer, viewTmplt); err != nil {
+			return err
 		}
+		lines = append(lines, buffer.String())
 	}
-
-	lines := strings.Split(buffer.String(), "\n")
-	// remove empy line at end
-	lines = lines[:len(lines)-1]
 	sort.Strings(lines)
-	fmt.Print(strings.Join(lines, "\n"))
+	fmt.Fprintln(writer, strings.Join(lines, "\n"))
```

Comment:
Actually, this does not seem to work when used with the '--format' option. My guess is that the buffer is not flushed in this case. This on the other hand works:

    	for _, line := range lines {
	 	fmt.Fprintln(writer, line)
	  }